[Bug][Docker][2.11] Docker build fails on Python Client Wheel Since Ubuntu Upgrade

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Version

Docker build of Pulsar 2.11 creating an all image.

### Minimal reproduce step

Perform a build of the all docker image.

### What did you expect to see?

Success

### What did you see instead?

Failure:

```
20:49:37  [INFO] Step 27/28 : RUN /pulsar/bin/install-pulsar-client.sh
20:49:37  [INFO] 
20:49:37  [INFO]  ---> Running in d98fec0c96bd
20:49:37  [INFO] ++ python3 -V
20:49:37  [INFO] ++ sed -E 's/.* ([[:digit:]]+)\.([[:digit:]]+).*/\1\2/'
20:49:37  [INFO] 
20:49:37  [INFO] + PYTHON_MAJOR_MINOR=310
20:49:37  [INFO] 
20:49:37  [INFO] ++ ls /pulsar/pulsar-client
20:49:37  [INFO] ++ grep cp310
20:49:37  [INFO] 
20:49:37  [INFO] + WHEEL_FILE=
20:49:37  [INFO] 
20:49:37  [INFO] + pip3 install '/pulsar/pulsar-client/[all]'
20:49:37  [INFO] 
20:49:38  [INFO] ERROR: Directory '/pulsar/pulsar-client/[all]' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.
20:49:38  [INFO] 
20:49:38  [ERROR] The command '/bin/sh -c /pulsar/bin/install-pulsar-client.sh' returned a non-zero code: 1
```

This error started happening after #20475 

### Anything else?

Either update the wheel creation to Python 3.10 (it's creating 3.8), or revert back to Ubuntu 20.04.

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR! to do the reversion.

--- a/docker/pulsar/pom.xml
+++ b/docker/pulsar/pom.xml
@@ -74,8 +74,8 @@
                   <workingDirectory>${project.basedir}/target</workingDirectory>
                   <executable>${project.basedir}/../../pulsar-client-cpp/docker/build-wheels.sh</executable>
                   <arguments>
-                    <!-- build python 3.8 -->
-                    <argument>3.8 cp38-cp38 manylinux2014 x86_64</argument>
+                    <!-- build python 3.10 -->
+                    <argument>3.10 cp310-cp310 manylinux2014 x86_64</argument>
                   </arguments>
                 </configuration>
               </execution>

